test/hotspot/jtreg/compiler/vectorization/TestSignumVector.java

```diff
@@ -82,11 +94,21 @@ public void test_signum_float(float[] fout, float[] finp) {
   public void kernel_test_round() {
       finp = new float[ARRLEN];
       fout = new float[ARRLEN];
+      Random rnd = new Random(20);
       for(int i = 0 ; i < ARRLEN; i++) {
-          finp[i] = (float)i*1.4f;
+          finp[i] = (i-ARRLEN/2)*rnd.nextFloat();
       }
       for (int i = 0; i < ITERS; i++) {
           test_signum_float(fout , finp);
       }
+      for(int i = 0 ; i < ARRLEN; i++) {
+        if (i-ARRLEN/2<0) {
+            if (fout[i] != -1.0)  throw new RuntimeException("Expected negative numbers in first half of array: " + java.util.Arrays.toString(fout));
+        } else if (i-ARRLEN/2==0) {
+            if (fout[i] != 0)     throw new RuntimeException("Expected zero in the middle of array: " + java.util.Arrays.toString(fout));
+        } else {
+            if (fout[i] != 1.0)   throw new RuntimeException("Expected positive numbers in second half of array: " + java.util.Arrays.toString(fout));
+        }
+    }
```

Comment:
Its ok to add correctness check here, but test only intend to perform check IR validations, there are detailed function tests in following files
test/hotspot/jtreg/compiler/intrinsics/math/TestSignumIntrinsic.java
test/hotspot/jtreg/compiler/c2/cr6340864/TestFloatVect.java
test/hotspot/jtreg/compiler/c2/cr6340864/TestDoubleVect.java

Reply:
I am ok to remove this change to the test.. I didn't know where the other tests where and by the time I did find those, already added this. (Figured "more test === good", but its just duplicate)

Reply:
I am find with it. Thanks!